02_configure_host.sh

```diff
@@ -37,7 +37,30 @@ fi
 # Allow ipmi to the virtual bmc processes that we just started
 sudo iptables -I INPUT -i virbr0 -p udp -m udp --dport 6230:6235 -j ACCEPT
 
+# Need to route traffic from the provisioning host.
+if [ "$EXT_IF" ]; then
+  sudo iptables -t nat -A POSTROUTING --out-interface $EXT_IF -j MASQUERADE
+  sudo iptables -A FORWARD --in-interface virbr0 -j ACCEPT
+fi
+
+# Need to pass the provision interface for bare metal
+if [ "$PRO_IF" ]; then
+  sudo ip link set "$PRO_IF" master brovc
+fi
+
+# Internal interface
+if [ "$INT_IF" ]; then
+  sudo ip link set "$PRO_IF" master virbr0
+fi
+
 # Switch NetworkManager to internal DNS
 sudo mkdir -p /etc/NetworkManager/conf.d/
 sudo crudini --set /etc/NetworkManager/conf.d/dnsmasq.conf main dns dnsmasq
-sudo systemctl restart NetworkManager
+if [ "$ADDN_DNS" ] ; then
+  echo "server=$ADDN_DNS" | tee /etc/NetworkManager/dnsmasq.d/upstream.conf
+fi
+if [[ "active" =~ $(systemctl is-active NetworkManager) ]]; then
```

Comment:
Minor nit: We don't actually need the string comparison/regex here I think, because systemctl is-active returns a non-zero status when it's not active.

Reply:
@hardys I think that is if i pass --quiet --. I thought if we hit a exit code other than 0 the script stops. That is why I was using regex...

Reply:
No the return code is the same regardless of -q/--quiet (although that's probably a good idea if you evaluate based on the return code vs the string).

The command/builtin bash is evaluating is "if", which doesn't return a non-zero code, even though it evaluates a code which does:

```
[shardy@tripleodev2 ~]$ systemctl is-active dmraid-activation
inactive
[shardy@tripleodev2 ~]$ echo $?
3
[shardy@tripleodev2 ~]$ systemctl is-active dmraid-activation -q
[shardy@tripleodev2 ~]$ echo $?
3
[shardy@tripleodev2 ~]$ systemctl is-active NetworkManager
active
[shardy@tripleodev2 ~]$ echo $?
0
[shardy@tripleodev2 ~]$ if systemctl is-active dmraid-activation -q; then echo "It is active"; else echo "It is not active"; fi
It is not active
[shardy@tripleodev2 ~]$ echo $?
0
```